src/layman/celery.py

```diff
@@ -35,28 +35,20 @@ def task_postrun(workspace, publication_type, publication_name, task_id, task_na
     key = LAST_TASK_ID_IN_CHAIN_TO_PUBLICATION
     hash = task_id
     if rds.hexists(key, hash):
-        finish_publication_chain(task_id, task_state)
         next_task = pop_step_to_run_after_chain(workspace, publication_type, publication_name)
+        update_related_publications_after_change(workspace, publication_type, publication_name)
+        finish_publication_chain(task_id, task_state)
```

Comment:
finish_publication_chain open the lock, and the "method" on line 45 expects the publication to be unlocked, so this order is needed.